src/nvim/terminal.c

```diff
@@ -618,6 +621,17 @@ static int term_movecursor(VTermPos new, VTermPos old, int visible,
   return 1;
 }
 
+// Set the terminal title. Takes ownership of the title string.
+static void buf_set_term_title(buf_T *buf, String title)
```

Comment:
Why is `title` of type `String` and not `const char*` or `const char_u*`? Looks like this argument is always a C-string before it is converted to `String`.